pkg/virt-api/webhooks/validating-webhook/admitters/instancetype-admitter.go

```diff
@@ -117,17 +128,73 @@ func validateRequestResource(ar *admissionv1.AdmissionReview, resourceType strin
 	return nil
 }
 
-func validateDedicatedCPUPlacement(instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) []metav1.StatusCause {
-	if instancetypeSpec == nil {
-		return nil
+func validateInstanceTypeCPU(instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) []metav1.StatusCause {
+	if instancetypeSpec != nil {
+		hasRequestedCPU := false
+		if instancetypeSpec.Resources != nil {
+			_, hasRequestedCPU = instancetypeSpec.Resources.Requests[k8sv1.ResourceCPU]
+		}
+
+		if instancetypeSpec.CPU == nil {
+			if !hasRequestedCPU {
```

Comment:
What about combining the two sets of conditionals here that check at least one of these is set into something like:

```
if (instancetypeSpec.CPU == nil || instancetypeSpec.CPU.Guest == 0) && !hasRequestedCPU {
```

You then just have the standalone check for both being defined.